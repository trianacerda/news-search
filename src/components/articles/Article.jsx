import React from 'react';
import PropTypes from 'prop-types';

export default function Article({ title, author, description }) {
  return (
    <figure>
      <figcaption>
        {title}
        {author}
        {description}
      </figcaption>
    </figure>
  );
}

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string,
  description: PropTypes.string.isRequired,
};
