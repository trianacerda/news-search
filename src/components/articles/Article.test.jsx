import React, { Component } from 'react';
import { render } from '@testing-library/react';
import Article from './Article';

describe('article test', () => {
  it('should render a single article for the list', () => {
    const { asFragment } = render(
      <Article
        title="triana loves candy"
        author="triana"
        description="triana goes to a candy shop"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
