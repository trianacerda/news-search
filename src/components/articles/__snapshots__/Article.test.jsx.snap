// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article test should render a single article for the list 1`] = `
<DocumentFragment>
  <figure>
    <figcaption>
      triana loves candytrianatriana goes to a candy shop
    </figcaption>
  </figure>
</DocumentFragment>
`;

exports[`should single article test should render a single article for the list 1`] = `
<DocumentFragment>
  <figure>
    <figcaption>
      triana loves candytrianatriana goes to a candy shop
    </figcaption>
  </figure>
</DocumentFragment>
`;

exports[`single article test should render a single article for the list 1`] = `
<DocumentFragment>
  <figure>
    <figcaption>
      triana loves candytrianatriana goes to a candy shop
    </figcaption>
  </figure>
</DocumentFragment>
`;
